### PR TITLE
fix 'passwd' field documentation

### PR DESCRIPTION
**Pull request by sshedi**
'passwd' field in cloud-config has no effect on an already existing
user. This was not documented earlier.

This change set adds that information to documentation.

LP: #1889545

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>

### CODE REVIEW

File: cloudinit/config/cc_users_groups.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
     - ``no_user_group``: Optional. Do not create group named after user.
       Default: false
     - ``passwd``: Hash of user password
+      Note: ``passwd`` has no effect on existing users.
```
Comment (OddBloke) [inline cloudinit/config/cc_users_groups.py:44]: As discussed in the bug, this isn't limited only to the `passwd` field: none of the configuration is applied for existing users. Could you move your note to below the configuration, alongside the other notes, and reword it to refer to all the configuration?

File: doc/examples/cloud-config-user-groups.txt [hunk 2]
```diff
@@ -59,6 +59,8 @@ users:
 #           (the above command would create from stdin an SHA-512 password hash
 #           with 4096 salt rounds)
 #
+#          This field has no effect on existing users.
+#
```
Comment (OddBloke) [inline doc/examples/cloud-config-user-groups.txt:63]: Similarly, this statement is not just applicable to the `passwd` field. Perhaps add some words after "Users are added after groups are added." further up in the file?

Comment (sshedi) [reply doc/examples/cloud-config-user-groups.txt]: Done, thanks.

File: cloudinit/config/cc_users_groups.py (hunk 1 above)
Comment (sshedi) [reply cloudinit/config/cc_users_groups.py]: @OddBloke - I have a doubt here. You are saying none of these configs apply on existing users but
```
ssh_authorized_keys
sudo
lock_passwd
ssh_redirect_user
```

By looking at code, the above configs seem to be applicable  on existing users? Please clarify the confusion.

Comment (OddBloke) [reply cloudinit/config/cc_users_groups.py]: Ah, yeah, you are correct, thanks for the pushback. I was getting mixed up between `add_user` and `create_user`. That's very confusing!

I think we should have a top-level message that notes that only those keys will be processed for existing users. What do you think?

Comment (sshedi) [reply cloudinit/config/cc_users_groups.py]: Agree, perhaps a note which says that following configs don't take effect on existing users or the other way round.
